packages/flutter_tools/lib/src/device.dart

```diff
@@ -490,6 +491,33 @@ abstract class Device {
     await descriptions(devices).forEach(globals.printStatus);
   }
 
+  static Future<void> printDevicesAsJson(List<Device> devices) async {
+    globals.printStatus(
+      const JsonEncoder.withIndent('  ').convert(
+        await Future.wait(devices.map((Device d) => d.toJson()))
+      )
+    );
+  }
+
+  Future<Map<String, Object>> toJson() async =>
```

Comment:
This method needs a doc comment, and a unit test for different device types.

Reply:
@jonahwilliams Could you please elaborate on "different device types"? What is the possible failure scenario to test against?

Reply:
You need to verify that your JSON mapping is correct. Suppose I changed key values randomly, that should make some test fail